### PR TITLE
Donates modal fix for Pc and mobile

### PR DESCRIPTION
**Pull request by Lolimkeri**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/app/common/components/modals/Donates/DonatesModal.component.tsx [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 const DonatesModal = () => {
     const { modalStore } = useMobx();
     const { setModal, modalsState: { donates } } = modalStore;
 
-    const [donateAmount, setDonateAmount] = useState(0);
+    const [donateAmount, setDonateAmount] = useState<number>(0);
+    const [donateName, setDonateName] = useState<string | undefined>('');
+    const [donateComment, setDonateComment] = useState<string | undefined>('');
+
     const [activeBtnIdx, setActiveBtnIndex] = useState<number>();
```
Comment (NadiaKishchuk) [inline src/app/common/components/modals/Donates/DonatesModal.component.tsx:25]: you can use a form from antd and then you will not need these hooks. You can also set the input type as a number or cat set the pattern for input, and when the submit button for the form is pressed, if the input doesn`t match, the submit will not be.

File: src/app/common/components/modals/Donates/DonatesModal.styles.scss [hunk 2]
```diff
@@ -102,9 +131,43 @@
...
+  
+        &:last-child {
+          font-family: 'Roboto';
+          font-style: normal;
+          font-weight: 300;
+          font-size: 15px;
+          line-height: 18px;
+          padding-inline-start: 16px;
+          height: 24px;
+          padding-top: 1vh;
```
Comment (Tysyatsky) [inline src/app/common/components/modals/Donates/DonatesModal.styles.scss:166]: use here our mixins for fonts and pxToRem

File: src/app/common/components/modals/Donates/DonatesModal.styles.scss [hunk 3]
```diff
@@ -156,58 +227,99 @@
...
+            vertical-align: middle;
+      
+            &:last-child {
+              height: 28px;
+              padding-top: 2px;
+              font-family: 'Roboto';
+              font-style: normal;
+              font-weight: 500;
+              font-size: 12px;
+              line-height: 14px;
```
Comment (Tysyatsky) [inline src/app/common/components/modals/Donates/DonatesModal.styles.scss:307]: and here

File: src/app/common/components/modals/Donates/DonatesModal.component.tsx [hunk 4]
```diff
@@ -31,15 +39,51 @@ const DonatesModal = () => {
...
+    var inputWidth = 0;
+    if(window.innerWidth>1024)
+    {
+        inputWidth = minWidth + donateAmount.toString().length * charWidth - count*firstWidth;
+    }
+    else
+    {
+        inputWidth = minWidth + donateAmount.toString().length * mobCharWidth - count*mobFirstWidth;
+    }
+    const style = { "--input-width": `${inputWidth}px` } as React.CSSProperties;
```
Comment (Tysyatsky) [inline src/app/common/components/modals/Donates/DonatesModal.component.tsx:72]: this looks like too much work for input width. Maybe u should try using useWindowSize hook for this task,

File: src/app/common/components/modals/Donates/DonatesModal.styles.scss (hunk 2 above)
Comment (Lolimkeri) [reply src/app/common/components/modals/Donates/DonatesModal.styles.scss]: fixed

File: src/app/common/components/modals/Donates/DonatesModal.styles.scss (hunk 3 above)
Comment (Lolimkeri) [reply src/app/common/components/modals/Donates/DonatesModal.styles.scss]: fixed

File: src/app/common/components/modals/Donates/DonatesModal.component.tsx (hunk 4 above)
Comment (Lolimkeri) [reply src/app/common/components/modals/Donates/DonatesModal.component.tsx]: changed logic

File: src/app/common/components/modals/Donates/DonatesModal.component.tsx [hunk 5]
```diff
@@ -53,38 +52,26 @@ const DonatesModal = () => {
...
     };
     
-    const minWidth = 0; // Minimum width of the input field
-    const charWidth = 42; // Width of each character in pixels
-    const firstWidth = 13;
-    const mobCharWidth = 21;
-    const mobFirstWidth = 6;
+    const charWidth = windowSize.width > 1024 ? 42 : 21;    // Width of each character in pixels
+    const firstWidth = windowSize.width > 1024 ? 13 : 6;
+
```
Comment (Katerix) [inline src/app/common/components/modals/Donates/DonatesModal.component.tsx:57]: perfect :)

**Review by Katerix — COMMENTED** [review]
good! fix import styles in DonatesModal.component.tsx and pr is ready to be merged!)

**Pull request merged**